src/type/validate.js

```diff
@@ -410,19 +425,41 @@ function validateObjectImplementsInterface(
     }
 
     // Assert additional arguments must not be required.
-    for (const objectArg of objectField.args) {
-      const argName = objectArg.name;
-      const ifaceArg = find(ifaceField.args, arg => arg.name === argName);
-      if (!ifaceArg && isRequiredArgument(objectArg)) {
+    for (const implementingArg of implementingField.args) {
+      const argName = implementingArg.name;
+      const implementedArg = find(
+        implementedField.args,
+        arg => arg.name === argName,
+      );
+      if (!implementedArg && isRequiredArgument(implementingArg)) {
         context.reportError(
-          `Object field ${object.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${iface.name}.${fieldName}.`,
-          [objectArg.astNode, ifaceField.astNode],
+          `Object field ${type.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${iface.name}.${fieldName}.`,
+          [implementingArg.astNode, implementedField.astNode],
         );
       }
     }
   }
 }
 
+function validateTypeImplementsAncestors(
+  context: SchemaValidationContext,
+  type: GraphQLObjectType | GraphQLInterfaceType,
+  iface: GraphQLInterfaceType,
+): void {
+  const implementingInterfaces = type.getInterfaces();
+  for (const transitive of iface.getInterfaces()) {
+    if (implementingInterfaces.indexOf(transitive) === -1) {
+      const typeKindStr = isObjectType(type) ? 'Object' : 'Interface';
```

Comment:
We have other similar errors and they simply refer to type as `Type` I don't think this error should be special:
https://github.com/graphql/graphql-js/blob/21de98ea4855df8c5146580611fda8ad76f52a2f/src/type/validate.js#L268
https://github.com/graphql/graphql-js/blob/21de98ea4855df8c5146580611fda8ad76f52a2f/src/type/validate.js#L325
Fixed in https://github.com/mike-marcacci/graphql-js/pull/1